cmd/kubectl-gadget/utils/trace.go

```diff
@@ -290,32 +455,158 @@ RetryLoop:
 		// Don't print warnings if at least one node succeeded. This avoids showing
 		// warnings together with the actual output generated by other nodes.
 		if successNodeCount == 0 {
-			printTraceFeedback(contextLogger.Warningf, nodeWarnings)
+			printWarningFunction := func(format string, args ...interface{}) { fmt.Printf(format, args) }
+
+			printTraceFeedback(printWarningFunction, nodeWarnings)
 		}
 
+		printErrorFunction := func(format string, args ...interface{}) { fmt.Fprintf(os.Stderr, format, args) }
+
 		// Print errors even if other nodes succeeded.
-		printTraceFeedback(contextLogger.Errorf, nodeErrors)
+		printTraceFeedback(printErrorFunction, nodeErrors)
 
 		if successNodeCount == 0 {
-			deleteTraces(contextLogger, traceRestClient, traceID)
-			contextLogger.Fatalf("Failed to run the gadget on all nodes: None of them succeeded")
+			return errors.New("Failed to run the gadget on all nodes: None of them succeeded")
 		}
+
 		break RetryLoop
 	}
 
+	client, err := k8sutil.NewClientsetFromConfigFlags(KubernetesConfigFlags)
+	if err != nil {
+		return fmt.Errorf("Error in creating setting up Kubernetes client: %w", err)
```

Comment:
```suggestion
		return fmt.Errorf("Error setting up Kubernetes client: %w", err)
```

Reply:
hey @eiffel-fl, I still see `Error in creating setting up Kubernetes client` there.